frontend/public/components/deploy-image.tsx

```diff
@@ -0,0 +1,439 @@
+                  <i className="fa fa-search" aria-hidden="true"></i>
+                  <span className="sr-only">Find</span>
+                </button>
+              </span>
+            </div>
+            <div className="help-block">
+              {/* TODO:  add link to create and image pull secret modal */}
+              To deploy an image from a private repository, you must <span>create an image pull secret</span> with your image registry credentials.
+            </div>
+          </div>
+          <div className="co-image-name-results">
+            {!isi && <div className="co-image-name-results__loading">
+              {loading && <Loading className="co-m-loader--inline" />}
+              {(!loading && !searchError) && <h2 className="h3 co-image-name-results__loading-heading">{/* Select an image stream tag or e*/}Enter an image name.</h2>}
+              {(!loading && searchError) && <React.Fragment>
+                <h2 className="h3 co-image-name-results__loading-heading">
+                  <i className="pficon pficon-error-circle-o" aria-hidden="true"></i> Could not load image metadata.
+                </h2>
+                <p className="co-image-name-results__loading-error">{searchError}</p>
+              </React.Fragment>}
+            </div>}
+            {isi && <div className="co-image-name-results__details">
+              <div className="row">
+                <div className="col-sm-3 col-md-2 hidden-xs text-right h2">
+                  <span className="fa fa-cube text-muted" style={{fontSize: '100px'}} aria-hidden="true"></span>
+                </div>
+                <div className="col-sm-9 col-md-10">
+                  {runsAsRoot(isi.image) && <div className="alert alert-warning">
+                    <span className="pficon pficon-warning-triangle-o" aria-hidden="true"></span>
+                    Image <strong>{isi.name}</strong> runs as the <strong>root</strong> user which might not be permitted by your cluster administrator.
+                  </div>}
+                  <h2 className="co-image-name-results__heading">
+                    {/* TODO:  add ability to import from image stream tag */}
+                    {/* <span ng-if="mode === 'dockerImage'">{isi.name}</span> */}
+                    {/* <span ng-if="mode === 'istag'">{istag.imageStream}<span ng-if="isi.tag">:{isi.tag}</span></span> */}
+                    {isi.name}
+                    <small>
+                      {_.get(isi, 'result.ref.registry') && <span>from {isi.result.ref.registry}, </span>}
+                      <React.Fragment><Timestamp timestamp={isi.image.dockerImageMetadata.Created} />, </React.Fragment>
+                      {_.get(isi, 'image.dockerImageMetadata.Size') && <span>{units.humanize(isi.image.dockerImageMetadata.Size, 'binaryBytes', true).string}, </span>}
+                      <React.Fragment>{isi.image.dockerImageLayers.length} layers</React.Fragment>
```

Comment:
I don't think you need React.Fragment.

Prefer _.size

Reply:
> Prefer _.size

Sorry.  I don't follow.  Instead of the second _.get?

Reply:
Instead of `.length` since it guards against undefined and works with objects